src/code.cloudfoundry.org/cf-tcp-router/config/config.go

```diff
@@ -262,28 +240,97 @@ func (c *Config) initConfigFromFile(path string, enableCertCreation bool) error
 	return nil
 }
 
-func certHasSAN(cert *x509.Certificate) bool {
-	hasSANExtension := false
+func (c *Config) createDir(dirPath string) error {
+	// ensure directory exists
+	if err := os.MkdirAll(dirPath, 0750); err != nil {
+		return err
+	}
+
+	targetUID, targetGID, err := c.getUIDGID()
+	if err != nil {
+		return err
+	}
+
+	// Change ownership
+	err = os.Chown(dirPath, targetUID, targetGID)
+	if err != nil {
+		return fmt.Errorf("Error changing ownership: %s", err)
+	}
+
+	return nil
+}
+
+func (c *Config) writeCertsToDisk(dirPath string, name string, cert FrontendTLSJob, privateKey string) error {
+	uid, gid, err := c.getUIDGID()
+	if err != nil {
+		return err
+	}
+
+	certFilePath := filepath.Join(dirPath, fmt.Sprintf("%s.pem", name))
+	keyFilePath := filepath.Join(dirPath, fmt.Sprintf("%s.pem.key", name))
+
+	if err := writeFile(certFilePath, []byte(cert.CertChain), 0750, uid, gid); err != nil {
+		return err
+	}
+
+	if err := writeFile(keyFilePath, []byte(privateKey), 0750, uid, gid); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Config) getUIDGID() (int, int, error) {
```

Comment:
is there anyway to tests this function?

Reply:
This function is looking up the user id and the group id. I am not sure if any part can be mocked out to create a useful test.

Thoughts?